dbt/adapters/clickhouse/impl.py

```diff
@@ -76,6 +78,15 @@ def get_clickhouse_cluster_name(self):
         if conn.credentials.cluster:
             return '"{}"'.format(conn.credentials.cluster)
 
+    @available
+    def get_clickhouse_version(self):
+        conn = self.connections.get_if_exists()
+        if isinstance(conn.handle, HttpClient):
+            return conn.handle.server_version
+        elif isinstance(conn.handle, ChNativeAdapter):
```

Comment:
Can you abstract this away into the ChNativeAdapter with a `server_version` @property field?